interview-questions.md

```diff
@@ -8,33 +8,54 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: 
 
-Researched answer:
+Object-oriented programming to me is a scripy that is focused more on extracting and mutating data from a class/object perspective focusing less on syntax. 
+
+Researched answer: 
+
+An efficient way of programming that lessen the complexity of how coding use to be. The idea is to have many parts working as one instead of one big block of code; which made errors harder to troubleshoot and effected more of the program as a whole.
 
 2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: 
+
+A Float is a decimal number data type and an integer is any whole number data type.
 
-Researched answer:
+Researched answer: 
+
+A Float is a decimal number data type and an integer is any whole number data type. A better way to describe a float is that it's a non-whole number. Floats also have to be inserted into the equation its never implied.
 
 3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: 
+
+Implicit return; meaning that the return is implied, or ruby automatically returns the data without extra syntax.
 
-Researched answer:
+Researched answer: 
+
+If no return is specified,the last line of the method is returned automatically.
 
 4. What is a block in Ruby?
 
-Your answer:
+Your answer: 
+
+A block in Ruby is a block of code from the name of the method to the end.
+
+Researched answer: 
 
-Researched answer:
+An anonymous function that is passed through a method to return th output.
 
 5. How do you extract a single value from a Ruby hash?
 
-Your answer:
+Your answer: 
+
+Im not too familiar with Ruby hashes, so I'll give what I might know about the process: Locating the desired value. I dont know if they're like arrays and you have to use indexes but I'm sure there are similarities.
+I'm remember Hash.new being a part of a ruby hash. And I think they're separated by the "arrow syntax", "=>" I think we call it a rocket or something. 
+
+Researched answer: 
 
-Researched answer:
+You can use the hash select method to extract a single value from a ruby rash.
```

Comment:
Remember that you can use bracket notation to extract a value from a hash. For example: `hash_name[:key_name]`

Reply:
Ok I need to do more studying on Ruby Hashes. I don't really understand what their purpose is at the moment. But I can research and give you feed back if you are ok with that.

Reply:
Of course!